ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StreamMapValuesBuilder.java

```diff
@@ -28,21 +29,26 @@ private StreamMapValuesBuilder() {
   public static <K> KStreamHolder<K> build(
       final KStreamHolder<K> stream,
       final StreamMapValues<K> step,
-      final KsqlQueryBuilder queryBuilder) {
+      final KsqlQueryBuilder queryBuilder
+  ) {
     final QueryContext queryContext = step.getProperties().getQueryContext();
+
     final LogicalSchema sourceSchema = step.getSource().getProperties().getSchema();
-    final Selection selection =
-        Selection.of(
-            queryBuilder.getQueryId(),
-            queryContext,
-            sourceSchema,
-            step.getSelectExpressions(),
-            queryBuilder.getKsqlConfig(),
-            queryBuilder.getFunctionRegistry(),
-            queryBuilder.getProcessingLogContext()
-        );
+
+    final SelectValueMapper<K> mapper = Selection.<K>of(
+        queryBuilder.getQueryId(),
+        queryContext,
+        sourceSchema,
+        step.getSelectExpressions(),
+        queryBuilder.getKsqlConfig(),
+        queryBuilder.getFunctionRegistry(),
+        queryBuilder.getProcessingLogContext()
+    ).getMapper();
+
+    final Named selectName = Named.as(queryBuilder.buildUniqueNodeName(step.getSelectNodeName()));
```

Comment:
Let's bake the uniqueness into the name we put into the step class. This way we don't have to worry about generating the same name each time (although I understand this doesn't matter so much for this change, since there's no state store). For example, if we do a projection before a join, we'd need to make sure to always build the join in the right order to keep this name consistent. Some alternatives here:
- incorporate the query context in the name, like we do for Joined/Grouped/Materialized
- store the generated name in the step instead of generating it here.

Reply:
I'll pick this up in a later PR if that's OK.